### PR TITLE
[react] Mark script's charSet attribute as deprecated

### PR DESCRIPTION
**Pull request by mbelsky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Attributes), [html standard](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by eps1lon — APPROVED** [review]

File: types/react/index.d.ts [hunk 1]
```diff
@@ -2236,12 +2236,23 @@ declare namespace React {
...
     interface ScriptHTMLAttributes<T> extends HTMLAttributes<T> {
         async?: boolean;
+        /** @deprecated */
         charSet?: string;
         crossOrigin?: string;
         defer?: boolean;
         integrity?: string;
         noModule?: boolean;
         nonce?: string;
+        referrerpolicy?:
```
Comment (mbelsky) [inline types/react/index.d.ts:2246]: It's lowercased because React does not recognize `referrerPolicy` prop on a DOM element

Comment (mbelsky) [reply types/react/index.d.ts:2246]: @eps1lon I've seen your comment in [another PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48580#discussion_r501087614) about lowercase attributes. Does it mean the `referrerpolicy` attribute should not be added?

Comment (eps1lon) [reply types/react/index.d.ts:2246]: We probably want to get this into React itself first if they don't already have `referrerPolicy`.

Comment (mbelsky) [reply types/react/index.d.ts]: Thanks for response. I've removed this change.

**Review by eps1lon — APPROVED** [review]

**Comment by johnnyreilly** [comment]
Thanks!

**Pull request merged**